src/test/java/org/wildfly/security/sasl/gs2/Gs2Test.java

```diff
@@ -524,54 +614,44 @@ private SaslClient getSaslClient(final String[] mechanisms, final String authori
         }
     }
 
-    private class Gs2ServerCallbackHandler implements CallbackHandler {
+    //TODO I don't like the indirect tests having to pass in a callback handler to get this information
+    private static class IndirectCallbackHandler implements CallbackHandler {
         private final String bindingType;
         private final byte[] bindingData;
 
-        public Gs2ServerCallbackHandler(final String bindingType, final byte[] bindingData) {
+        private IndirectCallbackHandler(String bindingType, byte[] bindingData) {
             this.bindingType = bindingType;
             this.bindingData = bindingData;
         }
 
+        @Override
         public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
             for (Callback callback : callbacks) {
                 if (callback instanceof ChannelBindingCallback) {
```

Comment:
This is a good point. It would be possible to set bindingType and bindingData in the properties that are passed in (by adding them to [WildFlySasl](https://github.com/wildfly-security/wildfly-elytron/blob/master/src/main/java/org/wildfly/security/sasl/WildFlySasl.java)). Perhaps we should ask David about his thoughts on this?